src/modules/0.0.23/aws_security_group/index.ts

```diff
@@ -140,31 +140,35 @@ class SecurityGroupMapper extends MapperBase<SecurityGroup> {
       return await client.deleteSecurityGroup(instanceParams);
     } catch (e: any) {
       if (e.Code === 'DependencyViolation') {
-        // Just wait for 5 min on every dependency violation and retry
-        await new Promise(resolve => setTimeout(resolve, 5 * 60 * 1000));
-        try {
-          return await client.deleteSecurityGroup(instanceParams);
-        } catch (e2: any) {
-          // If the dependency continues we add the dependency to the error message in order to
-          // debug what is happening
-          if (e2.Code === 'DependencyViolation') {
-            const sgEniInfo = await client.describeNetworkInterfaces({
-              Filters: [
-                {
-                  Name: 'group-id',
-                  Values: [`${instanceParams.GroupId}`],
-                },
-              ],
-            });
-            const eniMessage = `Network interfaces associated with security group ${
-              instanceParams.GroupId
-            }: ${JSON.stringify(sgEniInfo.NetworkInterfaces)}`;
-            e2.message = `${e2.message} | ${eniMessage}`;
+        const sgEniInfo = await client.describeNetworkInterfaces({
+          Filters: [
+            {
+              Name: 'group-id',
+              Values: [`${instanceParams.GroupId}`],
+            },
+          ],
+        });
+        // if the network interface is in detached state, we can remove it
+        if (sgEniInfo.NetworkInterfaces) {
+          for (const sg of sgEniInfo.NetworkInterfaces) {
+            if (sg.Attachment?.Status === 'detached') {
+              // we can just delete it
+              try {
+                await client.deleteNetworkInterface({ NetworkInterfaceId: sg.NetworkInterfaceId });
+                continue;
+              } catch (e) {
+                throw new Error('Error deleting network interface');
```

Comment:
It would be good to improve this message by adding more info to this error message, like for what sg it happened and maybe the network interface info.